### PR TITLE
[Merged by Bors] - feat: add extraContainers option to pods

### PR DESCRIPTION
**Pull request by nacardin**
Example values.yaml:
```
scPod:
  extraContainers:
  - name: my-sidecar
    image: my-sidecar:1.0.0
    imagePullPolicy: IfNotPresent
spuPod:
  extraContainers:
  - name: my-sidecar
    image: my-sidecar:1.0.0
    imagePullPolicy: IfNotPresent
```

### CODE REVIEW

**Review by sehz — APPROVED** [review]
LGTM

File: k8-util/helm/fluvio-app/templates/sc-deployment.yaml [hunk 1]
```diff
@@ -57,7 +57,10 @@ spec:
             - name: authorization-config
               mountPath: /etc/fluvio/authorization
             {{ end }}
+        {{ end }}
+        {{- toYaml .Values.scPod.extraContainers | nindent 8 }}
       volumes:
+      {{ if .Values.tls }}
```
Comment (morenol) [inline k8-util/helm/fluvio-app/templates/sc-deployment.yaml:63]: Shouldn't this be before `volumes:`?

Comment (nacardin) [reply k8-util/helm/fluvio-app/templates/sc-deployment.yaml:63]: works either way

Comment (morenol) [reply k8-util/helm/fluvio-app/templates/sc-deployment.yaml:63]: Do you know why the CI failed with:

```
        stderr: Error: INSTALLATION FAILED: YAML parse error on fluvio-app/templates/sc-deployment.yaml: error converting YAML to JSON: yaml: line 35: did not find expected '-' indicator
```

Comment (nacardin) [reply k8-util/helm/fluvio-app/templates/sc-deployment.yaml:63]: maybe it doesn't like the `extraContainers: []` placeholder. it worked before when I had an actual container there. trying to fix now

File: crates/fluvio-sc/src/k8/objects/spu_k8_config.rs [hunk 2]
```diff
@@ -25,6 +25,7 @@ pub struct PodConfig {
     pub resources: Option<ResourceRequirements>,
     pub storage_class: Option<String>,
     pub base_node_port: Option<u16>,
+    pub extra_containers: Vec<ContainerSpec>,
```
Comment (morenol) [inline crates/fluvio-sc/src/k8/objects/spu_k8_config.rs:28]: maybe #[serde(default) here?

File: k8-util/helm/fluvio-app/templates/sc-deployment.yaml (hunk 1 above)
Comment (nacardin) [reply k8-util/helm/fluvio-app/templates/sc-deployment.yaml:65]: fixed

**Review by morenol — APPROVED** [review]

**Comment by morenol** [comment]
bors retry

**Pull request closed without merging**